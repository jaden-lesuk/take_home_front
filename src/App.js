import React from 'react'
import axios from 'axios'
import 'bootstrap/dist/css/bootstrap.min.css'
import {Container, Button, Spinner} from 'react-bootstrap'

import Result from './components/Result'


class App extends React.Component{
  state ={
    city: "",
    weather: "*result will be here*"
  }

  onChange = (e) => this.setState({ [e.target.name] : e.target.value });

  onSubmit = async (e) => {
    e.preventDefault()
    this.setState({ weather : "" })
    await axios.post('https://flask-backend-testing.herokuapp.com/api', {city: this.state.city})
    .then(response => this.setState({ weather : response.data }))
  }

  render(){
    return (
      <div style={styles.mainBodyStyle}>
          <Container style={styles.contentStyle} fluid>
              <h1>How's the weather in?</h1>
              <form onSubmit={this.onSubmit}>
                <div className="mb-3">
                  <input 
                      type = "text"
                      name = "city"
                      className="form-control"
                      placeholder = "City ..."
                      value = {this.state.city}
                      onChange = { this.onChange }
                      required
                      />
                  </div>
                  <Button type="submit" size="md" variant="primary" block>Let's see...</Button>
               </form>
               </Container>

               {
                 this.state.weather === ""?
                  <div className="text-center"><Spinner animation="grow" variant="light" /></div>
                  : <Result weather={this.state.weather} />
               }
          
      </div>
    );
  }
}

const styles = {
  mainBodyStyle: {
    background: 'linear-gradient(to bottom, #3a1c71, #d76d77, #ffaf7b)',
    minHeight: '100vh',
    paddingTop: '10vh',
    
    paddingLeft: '15vw',
    paddingRight: '15vw',
  },
  contentStyle: {
    textAlign: 'center',
    color: 'white',
    marginBottom: "5vh",
  }
}

export default App;
