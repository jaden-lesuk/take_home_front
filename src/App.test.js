import React from 'react';
import App from './App';
import { create } from 'react-test-renderer'

describe('App Component Snapshot Test',()=>{
    test('testing app', () => {
    let tree = create(<App />)
    expect(tree.toJSON()).toMatchSnapshot();
  })
})
