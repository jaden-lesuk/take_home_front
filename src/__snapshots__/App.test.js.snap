// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Snapshot Test testing app 1`] = `
<div
  style={
    Object {
      "background": "linear-gradient(to bottom, #3a1c71, #d76d77, #ffaf7b)",
      "minHeight": "100vh",
      "paddingLeft": "15vw",
      "paddingRight": "15vw",
      "paddingTop": "10vh",
    }
  }
>
  <div
    className="container-fluid"
    style={
      Object {
        "color": "white",
        "marginBottom": "5vh",
        "textAlign": "center",
      }
    }
  >
    <h1>
      How's the weather in?
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div
        className="mb-3"
      >
        <input
          className="form-control"
          name="city"
          onChange={[Function]}
          placeholder="City ..."
          required={true}
          type="text"
          value=""
        />
      </div>
      <button
        className="btn btn-primary btn-block btn-md"
        disabled={false}
        type="submit"
      >
        Let's see...
      </button>
    </form>
  </div>
  <div
    className="text-center"
  >
    <div
      className="spinner-grow text-light"
    />
  </div>
</div>
`;
